src/matrices.rs

```diff
@@ -339,9 +446,9 @@ impl<T: MatrixNumber> Mul<Self> for Matrix<T> {
 
 impl<T: MatrixNumber> CheckedMul for Matrix<T> {
     fn checked_mul(&self, v: &Self) -> Option<Self> {
-        self.check_shape_for_mul(v).ok()?;
-        let (h, _) = self.get_shape().unwrap();
-        let (_, w) = v.get_shape().unwrap();
+        self.corresponding_shapes_for_mul(v).then_some(())?;
```

Comment:
What do you think about moving this `(h, w)` unpacking to `corresponding_shapes_for_mul` and making it return `Ok((h, w))` if shapes match or `Err` otherwise instead of bool? Then probably its name should be changed too.

Reply:
It is a good idea!